extensions/ql-vscode/src/data-extensions-editor/yaml.ts

```diff
@@ -142,3 +183,55 @@ export function createDataExtensionYaml(
   return `extensions:
 ${extensions.join("\n")}`;
 }
+
+export function loadDataExtensionYaml(
+  data: any,
+): Record<string, ModeledMethod> | undefined {
+  if (typeof data !== "object") {
+    return undefined;
```

Comment:
I'm wondering whether we should be throwing errors/showing warnings if the data is not valid, WDYT?

Reply:
I agree, but I think it's not essential for the initial functionality. I've created an internal issue linked to the batch so that we will add error handling later on.